SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -796,39 +798,58 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 	});
 }
 
-void USpatialActorChannel::SpatialViewTick()
+void USpatialActorChannel::ServerViewTick()
 {
-	if (Actor != nullptr && !Actor->IsPendingKill() && IsReadyForReplication())
+	if (!IsAuthoritativeServer())
+	{
+		return;
+	}
+
+	UObject* ControllerCDO = AController::StaticClass()->GetDefaultObject();
+	UObject* PlayerControllerCDO = APlayerController::StaticClass()->GetDefaultObject();
+
+	FString OldOwnerWorkerAttribute = NetOwnerWorkerAttribute;
+	NetOwnerWorkerAttribute = improbable::GetOwnerWorkerAttribute(Actor);
+
+	if (bFirstTick || OldOwnerWorkerAttribute != NetOwnerWorkerAttribute)
 	{
-		bool bOldNetOwned = bNetOwned;
+		bool bSuccess = Sender->UpdateEntityACLs(GetEntityId(), NetOwnerWorkerAttribute);
 
-		// Use Actor's connection to determine if client owned
-		bNetOwned = false;
-		if (UNetConnection* NetConnection = Actor->GetNetConnection())
+		if (bSuccess)
 		{
-			if (APlayerController* PlayerController = NetConnection->PlayerController)
-			{
-				bNetOwned = PlayerController->PlayerState != nullptr;
-			}
+			bFirstTick = false;
 		}
-
-		if (bFirstTick || bOldNetOwned != bNetOwned)
+		else
 		{
-			if (IsAuthoritativeServer())
-			{
-				bool bSuccess = Sender->UpdateEntityACLs(Actor, GetEntityId());
+			// Reset stored worker attribute so we retry setting ACLs next tick.
+			NetOwnerWorkerAttribute = OldOwnerWorkerAttribute;
+		}
+	}
+}
 
-				if (bFirstTick && bSuccess)
-				{
-					bFirstTick = false;
-				}
-			}
-			else if (!NetDriver->IsServer())
-			{
-				Sender->SendComponentInterest(Actor, GetEntityId());
+void USpatialActorChannel::ClientViewTick()
+{
+	bool bOldNetOwned = bNetOwned;
+	bNetOwned = IsOwnedByWorker();
 
-				bFirstTick = false;
-			}
+	if (bFirstTick || bOldNetOwned != bNetOwned)
+	{
+		Sender->SendComponentInterest(Actor, GetEntityId(), bNetOwned);
```

Comment:
Shouldn't the server being setting the component interest of the client? If this is just re-arranging stuff feel free to ignore.

Reply:
Is this still needed with QBI on?